transport/src/main/java/io/scalecube/transport/TransportImpl.java

```diff
@@ -195,6 +195,36 @@ public NetworkEmulator networkEmulator() {
                     "Failed to send {} to {}, cause: {}", message, address, ex.toString()));
   }
 
+  public Mono<Message> requestResponse(final Message request, Address address) {
+    return Mono.create(
+        sink -> {
+          Objects.requireNonNull(request, "request must be not null");
+          Objects.requireNonNull(request.correlationId(), "correlationId must be not null");
+
+          listen()
+              .filter(resp -> resp.correlationId() != null)
+              .filter(resp -> resp.correlationId().equals(request.correlationId()))
+              .take(1)
+              .subscribe(
+                  msg -> {
+                    sink.success(msg);
+                  },
+                  error -> {
+                    sink.error(error);
+                  });
+
+          send(address, request)
+              .subscribe(
+                  null,
+                  ex -> {
+                    LOGGER.warn(
+                        "Unexpected exception on transport request-response, cause: {}",
+                        ex.toString());
+                    sink.error(ex);
```

Comment:
We could not send and notify sink with error. Great. But we not cleaning up resources when we couldn't send -- at line 208 (or line 204). I propose to dispose previously created result of listen().subscribe() when we couldn't send a msg.